DataFormats/ForwardDetId/interface/BTLDetId.h

```diff
@@ -18,20 +18,45 @@ class BTLDetId : public MTDDetId {
   
  private:
   
-  static const uint32_t kBTLmoduleOffset           = 10;
-  static const uint32_t kBTLmoduleMask             = 0x3F;
-  static const uint32_t kBTLmodTypeOffset          = 8;
-  static const uint32_t kBTLmodTypeMask            = 0x3;
-  static const uint32_t kBTLCrystalOffset          = 0;
-  static const uint32_t kBTLCrystalMask            = 0x3F;
+  static constexpr uint32_t kBTLmoduleOffset           = 10;
+  static constexpr uint32_t kBTLmoduleMask             = 0x3F;
+  static constexpr uint32_t kBTLmodTypeOffset          = 8;
+  static constexpr uint32_t kBTLmodTypeMask            = 0x3;
+  static constexpr uint32_t kBTLCrystalOffset          = 0;
+  static constexpr uint32_t kBTLCrystalMask            = 0x3F;
+
+ public:
+
+  /// range constants, need two sets for the time being (one for tiles and one for bars)
```

Comment:
I am worried about putting this much geometry info into the DetId. It seems like any of these numbers could change, in principle. It might be better to put this functionality into a topology class.

Reply:
This is functional to the definition of a hashed detId, it is basically a clone of the approach we are using in EBDetId since more than a decade (from a logic point of view the two detectors can be treated in the same way). Otherwise an ancillary class would be needed for different definitions of DetId, and I think that such an object should be as self contained as possible. 
I see several "magic numbers" embedded in HcalDetId, and also in MuonDetId there are limits...